docs/tutorials/adguardhome.md

```diff
@@ -0,0 +1,194 @@
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: external-dns
+rules:
+- apiGroups: [""]
+  resources: ["services","endpoints","pods"]
+  verbs: ["get","watch","list"]
+- apiGroups: ["extensions","networking.k8s.io"]
+  resources: ["ingresses"]
+  verbs: ["get","watch","list"]
+- apiGroups: [""]
+  resources: ["nodes"]
+  verbs: ["list","watch"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: external-dns-viewer
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: external-dns
+subjects:
+- kind: ServiceAccount
+  name: external-dns
+  namespace: default
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: external-dns
+spec:
+  strategy:
+    type: Recreate
+  selector:
+    matchLabels:
+      app: external-dns
+  template:
+    metadata:
+      labels:
+        app: external-dns
+    spec:
+      serviceAccountName: external-dns
+      containers:
+      - name: external-dns
+        image: registry.k8s.io/external-dns/external-dns:v0.13.5
+        # If authentication is disabled and/or you didn't create
+        # a secret, you can remove this block.
+        envFrom:
+        - secretRef:
+            # Change this if you gave the secret a different name
+            name: adguardhome-password
+        args:
+        - --source=service
+        - --source=ingress
+        # AdGuard Home only supports A/AAAA/CNAME records so there is no mechanism to track ownership.
+        # You don't need to set this flag, but if you leave it unset, you will receive warning
+        # logs when ExternalDNS attempts to create TXT records.
+        - --registry=noop
+        # IMPORTANT: If you have records that you manage manually in AdGuard Home, set
+        # the policy to upsert-only so they do not get deleted.
+        - --policy=upsert-only
+        - --provider=adguardhome
+        # Change this to the actual address of your AdGuard Home web server
+        - --adguardhome-server=http://adguardhome.dns.svc.cluster.local
+      securityContext:
+        fsGroup: 65534 # For ExternalDNS to be able to read Kubernetes token files
+```
+
+### Arguments
+
+ - `--adguardhome-server (env: EXTERNAL_DNS_ADGUARDHOME_SERVER)` - The address of the AdGuard Home web server
+ - `--adguardhome-username (env: EXTERNAL_DNS_ADGUARDHOME_USERNAME)` - The username to the AdGuard Home web server
+ - `--adguardhome-password (env: EXTERNAL_DNS_ADGUARDHOME_PASSWORD)` - The password to the AdGuard Home web server
+ - `--adguardhome-tls-skip-verify (env: EXTERNAL_DNS_ADGUARDHOME_TLS_SKIP_VERIFY)` - Skip verification of any TLS certificates served by the AdGuard Home web server.
+
+## Verify ExternalDNS Works
+
+### Ingress Example
+
+Create an Ingress resource. ExternalDNS will use the hostname specified in the Ingress object.
+
+```yaml
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: foo
+spec:
+  ingressClassName: nginx
```

Comment:
this does not matter

Reply:
Basically, copied and pasted from pi hole provider docs. I've removed it :)